substrate/utils/frame/benchmarking-cli/src/extrinsic/bench.rs

```diff
@@ -146,7 +153,15 @@ where
 		let ext_builder = if let Some(ext_builder) = ext_builder {
 			ext_builder
 		} else {
-			return Ok((builder.build()?.block, None))
+			let proof_size = builder.current_proof_size();
+			return Ok((
```

Comment:
I'd personally prefer to skip pushing transactions to block and have common return path. Again it seems that some little parts of code are duplicated.